motr/io_nw_xfer.c

```diff
@@ -736,6 +751,52 @@ static void irfop_fini(struct ioreq_fop *irfop)
 	M0_LEAVE();
 }
 
+static int m0_buf_from_bufvec(struct m0_buf *dest,
+		                    const struct m0_bufvec *src)
+{
+	int i;
+	int count = 0;
+	int len_seg = 0;
+	void *dst;
+
+	M0_SET0(dest);
+
+	if (src == NULL)
+		return 0;
+
+	if (src->ov_vec.v_nr == 0)
+		return 0;
+
+	count = m0_vec_count(&src->ov_vec);
+	//YJC_TODO : Need to free buffer
+	dest->b_addr = m0_alloc(count);
+	if (dest->b_addr == NULL)
+		return M0_ERR(-ENOMEM);
+	dst = dest->b_addr;
+	for (i = 0; i < src->ov_vec.v_nr; ++i) {
+		char str[64];
+		len_seg = src->ov_vec.v_count[i];
+		if (len_seg > 0) {
+			snprintf(str, len_seg, "%s", (char *)src->ov_buf[i]);
+			//YJC_TODO: length shoudl len_seg instead of len_seg -1
+			// this is only for debug because we have string 
+			memcpy(dst, src->ov_buf[i], len_seg - 1);
```

Comment:
memcpy to dst will wipe off previously written data.

dst + written_till_now something like this should happen.